src/jquery.uls.lcd.js

```diff
@@ -22,43 +22,40 @@
 ( function ( $ ) {
 	'use strict';
 
-	var noResultsTemplate, LanguageCategoryDisplay;
-
-	noResultsTemplate = $( '<div>' ).addClass( 'uls-no-results-view hide' );
-	noResultsTemplate.append(
-		$( '<h2>' )
-			.attr( 'data-i18n', 'uls-no-results-found' )
-			.addClass( 'uls-no-results-found-title' )
-			.text( 'No results found' ),
-		$( '<div>' )
-			.addClass( 'uls-no-found-more' )
-			.append(
-				$( '<div>' )
-					.addClass( '' )
-					.append(
-						$( '<p>' ).append(
-							$( '<span>' )
-								.attr( 'data-i18n', 'uls-search-help' )
-								.text( 'You can search by language name, script name, ISO code of language or you can browse by region.' )
-						)
-					)
-			)
-	);
-
-	LanguageCategoryDisplay = function ( element, options ) {
+	// eslint-disable-next-line no-multi-str
+	var noResultsTemplate = '<div class="uls-no-results-view"> \
+		<h2 data-i18n="uls-no-results-found" class="uls-no-results-found-title">No results found</h2> \
+		<div class="uls-no-found-more"> \
+		<div><p> \
+		<span data-i18n="uls-search-help">You can search by language name, script name, ISO code of language or you can browse by region.</span>\
+		</p></div> \
+		</div></div>';
+
+	/**
+	 * Language category display
+	 * @param {Element} element The container element to which the languages to be displayed
+	 * @param {Object} [options] Configuration object
+	 * @cfg {Object} [languages] Languages known to the ULS. Keyed by language code, values are autonyms.
+	 * @cfg {string[]} [showRegions] Array of region codes to show. Default is
+	 *  [ 'WW', 'AM', 'EU', 'ME', 'AF', 'AS', 'PA' ],
+	 * @cfg {number} [itemsPerColumn] Number of languages per column.
+	 * @cfg {number} [columns] Number of columns for languages. Default is 4.
+	 * @cfg {function} [languageDecorator] Callback function to be called when a language
+	 *  link is prepared - for custom decoration.
+	 * @cfg {function|string[]} [quickList] The languages to display as suggestions for quick selectoin.
+	 * @cfg {string|jQuery} [noResultsTemplate]
```

Comment:
Let's not accept strings. It's a bad practice, as there isn't sufficient escaping. I predict we will need to accept functions here so that we can pass parameters (for example to differentiate between no matches or unknown language).

Reply:
I improved the code to accept functions and passes the current search query as param